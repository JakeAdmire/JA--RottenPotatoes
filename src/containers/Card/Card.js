import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import { setCard } from '../../actions';
import { connect } from 'react-redux';

export class Card extends Component {

  buildTitle = (name) => {
    let nameArray = name.toLowerCase().split(' ');

    return nameArray.map(word => {
      return word.charAt(0).toUpperCase() + word.slice(1);
    }).join(' ');
  }

  buildAddress = (address) => {
    let addressArray = address.toLowerCase().split(' ');

    addressArray = addressArray.map(word => {
      return word.charAt(0).toUpperCase() + word.slice(1);
    }).reverse();

    const bigCity = addressArray[1].toUpperCase();
    addressArray.splice(1, 1, bigCity);

    return addressArray.reverse().join(' ');
  }

  buildDistance = (distance) => {
    let splitDistance = distance.toString().split('.');
    let decimal = splitDistance[1].slice(0, 1);

    return splitDistance[0] + '.' + decimal + 'm away';
  }

  handleClick = () => {
    this.props.setCard(this.props);
  }

  render() {
    const { name, address, distance, id } = this.props;
    const { inspectionscore, violationtype } = this.props.correctPlaces[0];

    let headerText = this.buildTitle(name);
    let addressText = this.buildAddress(address);
    let distanceText = this.buildDistance(distance);

    return (
      <div className="Card">
        <div className="card-text">
          <Link to={`/locations/${id}`} onClick={this.handleClick}>{headerText}</Link>
          <p>{addressText}</p>
          <p>{distanceText}</p>
        </div>
        <div className="card-percentage">
          <p>inspection score:</p>
          <p>{ inspectionscore + '%' }</p>
          <p>{ violationtype }</p>
        </div>
        <div className="card-favorite">
          +
        </div>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setCard: (card) => dispatch(setCard(card))
})

export default connect(null, mapDispatchToProps)(Card);