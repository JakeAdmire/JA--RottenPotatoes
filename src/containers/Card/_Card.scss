.Card {
  background: #4D4D4D;
  color: white;
  margin: 10px auto;
  width: 580px; height: 150px;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: 1fr 165px 66px;

  .card-text {
    padding: 10px;
    position: relative;

    h3 { 
      font-family: 'Lobster', cursive; 
      width: 100%;
      margin: 0;
      font-size: 30px;
      text-decoration: underline;
    }

    p { 
      color: #BFBFBF; 
      font-size: 20px;
      margin: 10px 0 0;
    }

    p:nth-child(2) { width: 325px; }

    p:nth-child(3) {
      position: absolute;
      bottom: 10px; right: 0;
      font-size: 14px;
    }
  }

  .card-percentage {
    height: 100%;
    background: #BFBFBF;
    box-sizing: border-box;
    border: solid #4d4d4d 10px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
    font-weight: bold;
    font-size: 18px;

    p { margin: 0; }

    p:nth-child(2) {
      font-size: 55px;
    }
  }

  .card-favorite {
    height: 100%;
    background: #666666;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 80px;
    font-weight: bold;
  }
}