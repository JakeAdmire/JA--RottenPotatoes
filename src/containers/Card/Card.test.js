import React from 'react';
import { shallow } from 'enzyme';

import { Card } from './Card';

describe('Card', () => {

  let wrapper;
  const mockProps = {
    name: 'TACO BELL',
    address: '123 STREET ST. DENVER, CO',
    distance: 23.2113,
    id: 1,
    inspectionscore: 25,
    violationtype: 'critical',
    correctPlaces: [{}, {}, {}]
  }

  beforeEach(() => {
    wrapper = shallow (
      <Card {...mockProps} />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should return the proper value when buildTitle is invoked', () => {
    let buildTitle = wrapper.instance().buildTitle(mockProps.name);
    expect(buildTitle).toEqual("Taco Bell");
  })

  it('should return the proper value when buildDistance is invoked', () => {
    let buildDistance = wrapper.instance().buildDistance(mockProps.distance);
    expect(buildDistance).toEqual("23.2m away");
  })

  it('should return the proper value when buildAddress is invoked', () => {
    let buildAddress = wrapper.instance().buildAddress(mockProps.address);
    expect(buildAddress).toEqual("123 Street St. DENVER, Co");
  })
  
})