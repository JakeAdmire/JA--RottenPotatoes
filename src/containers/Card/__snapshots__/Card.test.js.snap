// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    address="123 STREET ST. DENVER, CO"
    correctPlaces={
      Array [
        Object {},
        Object {},
        Object {},
      ]
    }
    distance={23.2113}
    id={1}
    inspectionscore={25}
    name="TACO BELL"
    setCard={[MockFunction]}
    violationtype="critical"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-text"
        >
          <Link
            onClick={[Function]}
            to="/Rotten-Potatoes/locations/1"
          >
            Taco Bell
          </Link>
          <p>
            123 Street St. DENVER, Co
          </p>
          <p>
            23.2m away
          </p>
        </div>,
        <div
          className="card-percentage"
        >
          <p>
            inspection score:
          </p>
          <p>
            undefined%
          </p>
          <p />
        </div>,
        <div
          className="card-favorite"
        >
          +
        </div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              onClick={[Function]}
              to="/Rotten-Potatoes/locations/1"
            >
              Taco Bell
            </Link>,
            <p>
              123 Street St. DENVER, Co
            </p>,
            <p>
              23.2m away
            </p>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Taco Bell",
              "onClick": [Function],
              "to": "/Rotten-Potatoes/locations/1",
            },
            "ref": null,
            "rendered": "Taco Bell",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "123 Street St. DENVER, Co",
            },
            "ref": null,
            "rendered": "123 Street St. DENVER, Co",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "23.2m away",
            },
            "ref": null,
            "rendered": "23.2m away",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              inspection score:
            </p>,
            <p>
              undefined%
            </p>,
            <p />,
          ],
          "className": "card-percentage",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "inspection score:",
            },
            "ref": null,
            "rendered": "inspection score:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "undefined%",
            },
            "ref": null,
            "rendered": "undefined%",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "card-favorite",
        },
        "ref": null,
        "rendered": "+",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-text"
          >
            <Link
              onClick={[Function]}
              to="/Rotten-Potatoes/locations/1"
            >
              Taco Bell
            </Link>
            <p>
              123 Street St. DENVER, Co
            </p>
            <p>
              23.2m away
            </p>
          </div>,
          <div
            className="card-percentage"
          >
            <p>
              inspection score:
            </p>
            <p>
              undefined%
            </p>
            <p />
          </div>,
          <div
            className="card-favorite"
          >
            +
          </div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                onClick={[Function]}
                to="/Rotten-Potatoes/locations/1"
              >
                Taco Bell
              </Link>,
              <p>
                123 Street St. DENVER, Co
              </p>,
              <p>
                23.2m away
              </p>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Taco Bell",
                "onClick": [Function],
                "to": "/Rotten-Potatoes/locations/1",
              },
              "ref": null,
              "rendered": "Taco Bell",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "123 Street St. DENVER, Co",
              },
              "ref": null,
              "rendered": "123 Street St. DENVER, Co",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "23.2m away",
              },
              "ref": null,
              "rendered": "23.2m away",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                inspection score:
              </p>,
              <p>
                undefined%
              </p>,
              <p />,
            ],
            "className": "card-percentage",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "inspection score:",
              },
              "ref": null,
              "rendered": "inspection score:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "undefined%",
              },
              "ref": null,
              "rendered": "undefined%",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "card-favorite",
          },
          "ref": null,
          "rendered": "+",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
